obstacle_detector:
  ros__parameters:
    # Topic parameters
    lidar_points_topic: "/ouster/points"
    cloud_ground_topic: "/cloud/ground"
    cloud_clusters_topic: "/cloud/clusters"
    vision_bboxes_topic: "/bboxes/vision"
    autoware_objects_topic: "/detected_objects"
    bbox_target_frame: "base_link"

    # LiDAR Obstacle Detector Configuration
    use_pca_box: true  # Enable PCA for bounding boxes? (Activer PCA pour les boîtes englobantes ?) 
    use_tracking: true   # Enable obstacle tracking? (Activer le suivi des obstacles ?)

    voxel_grid_size: 0.06  # Voxel grid filter size to downsample point cloud (Taille du filtre voxel pour réduire le nuage de points)

    # Region of Interest (ROI) - Définition de la zone d'intérêt
    roi_max_x: 10.0    # Max detection distance in X (longueur max de détection en X)
    roi_max_y: 10.0    # Max detection width in Y (largeur max de détection en Y)
    roi_max_z: 5.0     # Max detection height in Z (hauteur max de détection en Z)
    roi_min_x: -20.0   # Min detection distance in X (longueur min de détection en X)
    roi_min_y: -20.0   # Min detection width in Y (largeur min de détection en Y)
    roi_min_z: -4.0  # Min detection height in Z (hauteur min de détection en Z, pour filtrer le sol)

    ground_threshold: 0.3  # Max height for a point to be considered ground (Hauteur max pour être considéré comme sol)

    # Clustering Parameters - Paramètres de regroupement des obstacles
    cluster_threshold: 0.6    # Max distance between points to form a cluster (Distance max entre deux points pour former un groupe)
    cluster_max_size: 5000    # Max number of points in a cluster (Nombre max de points par cluster)
    cluster_min_size: 10      # Min number of points to form a valid cluster (Nombre min de points pour un cluster valide)

    # Tracking Parameters - Paramètres de suivi des obstacles
    displacement_threshold: 1.0  # Max movement allowed per frame for tracking (Déplacement max autorisé par frame pour le suivi)
    iou_threshold: 1.0           # IoU threshold for matching tracked objects (Seuil IoU pour associer les objets suivis)

    # Visualization Parameters - Paramètres de visualisation